columnflow/tasks/framework/mixins.py

```diff
@@ -2157,8 +1936,173 @@ def processes_repr(self):
+                        config_inst.x("dataset_groups", {}),
+                    ))
+
+            elif "processes" in params:
+                # pick all datasets that contain any of the requested (sub) processes
+                datasets = list()
+                for i, _processes in enumerate(processes):
+
+                    sub_process_insts = sum((
+                        [proc for proc, _, _ in process_inst.walk_processes(include_self=True)]
+                        for process_inst in map(config_insts[i].get_process, _processes)
+                    ), [])
+                    datasets.append(tuple(
+                        dataset_inst.name
+                        for dataset_inst in config_insts[i].datasets
+                        if any(map(dataset_inst.has_process, sub_process_insts))
+                    ))
+
+            # complain when no datasets were found
+            if not datasets and not cls.allow_empty_datasets:
+                raise ValueError(f"no datasets found matching {params['datasets']}")
+
+            params["datasets"] = tuple(datasets)
+            params["dataset_insts"] = tuple(
+                tuple(config_inst.get_dataset(d) for d in params["datasets"][i])
+                for i, config_inst in enumerate(config_insts)
+            )
+            params["configs_vs_dataset_insts"] = dict(zip([c.name for c in config_insts], params["dataset_insts"]))
+
+        return params
+
+    @classmethod
+    def get_known_shifts(cls, config_inst, params):
+        shifts, upstream_shifts = super().get_known_shifts(config_inst, params)
+
+        # add shifts of all datasets to upstream ones
+        for dataset_inst in params.get("configs_vs_dataset_insts")[config_inst.name] or []:
+            if dataset_inst.is_mc:
+                upstream_shifts |= set(dataset_inst.info.keys())
+
+        return shifts, upstream_shifts
+
+    @property
+    def datasets_repr(self):
+        datasets = self.datasets
+        if len(set(datasets)) == 1:
+            datasets = (self.datasets[0],)
+
+        if (len(self.datasets) == 1) & (len(self.datasets[0]) == 1):
+            return self.datasets[0][0]
+
+        _repr = ""
+        merge_datasets = []
+        for _datasets in sorted(datasets):
+            merge_datasets += sorted(_datasets)
+            _repr += f"{len(_datasets)}_"
+        return _repr + f"_{law.util.create_hash(sorted(merge_datasets))}"
+
+    @property
+    def processes_repr(self):
+        processes = self.processes
+        if len(set(processes)) == 1:
+            processes = (self.processes[0],)
+
+        if (len(self.processes) == 1) & (len(self.processes[0]) == 1):
+            return self.processes[0][0]
+
+        _repr = ""
+        merge_processes = []
+        for _processes in sorted(processes):
+            merge_processes += sorted(_processes)
+            _repr += f"{len(_processes)}_"
+        return _repr + f"_{law.util.create_hash(sorted(merge_processes))}"
```

Comment:
This is copy-paste of basically the same function, I think this could be centralized for better maintainability